jib-core/src/main/java/com/google/cloud/tools/jib/ncache/CacheReadEntry.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.ncache;
+
+import com.google.cloud.tools.jib.blob.Blob;
+import com.google.cloud.tools.jib.image.DescriptorDigest;
+import java.util.Optional;
+
+/**
+ * Represents a cache entry for a layer stored in the cache. <b>Implementations must be
+ * immutable.</b>
+ */
+public interface CacheReadEntry {
```

Comment:
How about `CachedLayer`?

The design doc talks of caching layers and metadata, and so this type should seem to do both.  But it seems specific to layers.  And what's an entry?

Reply:
The original cache defined `CachedLayer`, which implemented the `Layer` interface. However, the coupling with the `Layer` interface meant that there was a lot of extra logic and code to support `CachedLayer` in places where `Layer` would be used. Rather, I think with the new cache interfaces, it'd be better to keep the cache entry decoupled from the `Layer` interface. A `Layer` instance can be constructed from a cache entry, but instead of having to use the subclass `CachedLayer` in many places like in the old cache, we can just use `Layer` everywhere. **I think maybe `CacheEntry` could work as a name?** (removing the `Read`)